ansible/roles/go-cli/tasks/deploy.yml

```diff
@@ -6,15 +6,39 @@
     path: "{{ go_cli_conf_dir }}"
     state: directory
 
+- name: "Ensure nginx directory for go-cli exists"
+  file:
+    path: "{{ go_cli_nginx_dir  }}"
+    state: directory
+
 - name: "Pull the {{ docker_image_tag }} image of go-cli"
   shell: "docker pull {{ docker_registry }}whisk/go-cli:{{ docker_image_tag }}"
   when: docker_registry != ""
 
-# TODO: Publish compressed binaries using nginx
+- name: "Build Docker image"
+  shell: "{{ openwhisk_home }}/tools/docker/dockerWithRetry.sh 240 build --build-arg BUILDTIME={{ whisk.version.date }} -t whisk/go-cli {{ openwhisk_home }}/tools/go-cli"
+
+- name: "Run Docker"
+  shell: "docker run -d --name go-cli whisk/go-cli"
+
+- name: Copy all the go-cli mac binaries into nginx
+  shell: "docker cp go-cli:/src/github.com/go-whisk-cli/mac '{{ go_cli_nginx_dir }}'"
+  ignore_errors: yes
+
+- name: Copy all the go-cli linux binaries into nginx
+  shell: "docker cp go-cli:/src/github.com/go-whisk-cli/linux '{{ go_cli_nginx_dir }}'"
+  ignore_errors: yes
+ 
+- name: Copy all the go-cli windows binaries into nginx
+  shell: "docker cp go-cli:/src/github.com/go-whisk-cli/windows '{{ go_cli_nginx_dir }}'"
+  ignore_errors: yes
 
-- name: remove go-cli container after publishing
+- name: Remove go-cli container after publishing
```

Comment:
why do you need 2 steps here: one with docker module and one with docker command to remove the `go-cli` container?